src/frontends/onnx/tests/onnx_editor_topological_sort.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+
+#include "common_test_utils/file_utils.hpp"
+#include "editor.hpp"
+#include "engines_util/test_case.hpp"
+#include "gtest/gtest.h"
+#include "ngraph/file_util.hpp"
+#include "onnx_test_util.hpp"
+#include "util/test_control.hpp"
+
+using namespace ov;
+using namespace ov::onnx_editor;
+using namespace ngraph::test;
+
+static std::string s_manifest = "${MANIFEST}";
+
+NGRAPH_TEST(onnx_editor, topological_sort_two_nodes_swap) {
+    ONNXModelEditor editor{ngraph::file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                        SERIALIZED_ZOO,
+                                                        "onnx/model_editor/topological_sort/two_nodes_swap.onnx")};
+    ASSERT_NO_THROW(editor.get_function());
+}
+
+NGRAPH_TEST(onnx_editor, topological_sort_completely_unsorted) {
+    ONNXModelEditor editor{ngraph::file_util::path_join(CommonTestUtils::getExecutableDirectory(),
+                                                        SERIALIZED_ZOO,
+                                                        "onnx/model_editor/topological_sort/completely_unsorted.onnx")};
+    editor.get_function();
```

Comment:
Missed assertion added ;)
Additional tests for multi-output and Constant node also added.
I'm skipping only a comparison graph if sorting is applied, because there can be more than one correct result of graph sorting. Of course the current implementation give you the same result for each call, but I'd rather prefer to avoid hardcoding some pattern of expected result.